ethexe/processor/src/handling/events.rs

```diff
@@ -0,0 +1,247 @@
+use anyhow::Result;
+use ethexe_common::{
+    mirror::RequestEvent as MirrorEvent,
+    router::{RequestEvent as RouterEvent, ValueClaim},
+    wvara::RequestEvent as WVaraEvent,
+};
+use ethexe_db::CodesStorage;
+use ethexe_runtime_common::state::{ComplexStorage as _, Dispatch, Storage};
+use gear_core::{
+    ids::ProgramId,
+    message::{DispatchKind, SuccessReplyReason},
+};
+use gprimitives::{ActorId, CodeId, MessageId, H256};
+use std::collections::BTreeMap;
+
+use crate::Processor;
+
+impl Processor {
+    pub(crate) fn handle_router_event(
+        &mut self,
+        states: &mut BTreeMap<ProgramId, H256>,
+        event: RouterEvent,
+    ) -> Result<()> {
+        match event {
+            RouterEvent::ProgramCreated { actor_id, code_id } => {
+                self.handle_new_program(actor_id, code_id)?;
+
+                states.insert(actor_id, H256::zero());
+            }
+            RouterEvent::CodeValidationRequested { .. }
+            | RouterEvent::BaseWeightChanged { .. }
+            | RouterEvent::StorageSlotChanged
+            | RouterEvent::ValidatorsSetChanged
+            | RouterEvent::ValuePerWeightChanged { .. } => {
+                log::debug!("Handler not yet implemented: {event:?}");
+                return Ok(());
+            }
```

Comment:
As I understand this events must be skipped by processor, so this debug should be removed

Reply:
It should be processed, but not yet implemented

Reply:
CodeValidationRequested - no

Reply:
will be processed here, once replaced with local event that contains code

Reply:
having processing of codes out of processor is incorrect (current behaviour)